### DISCUSSION TITLE
Default Node Resource For Container Without Limit

### DISCUSSION THREAD

**Discussion by zhoub** [k3s-io/k3s#4186] [General]
Hi,

Just would like to confirm that, if k8s would assign all the resources to a Pod at Node if no limit was set ?

According the official document,
https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/#if-you-specify-a-cpu-limit-but-do-not-specify-a-cpu-request
it should be.

Thanks in advance.

**Comment by brandond** [comment]
I'm not sure what you mean by "all" the resources. The docs are pretty clear:

> If you specify a CPU limit for a Container but do not specify a CPU request, Kubernetes automatically assigns a CPU request that matches the limit. 
> Similarly, if a Container specifies its own memory limit, but does not specify a memory request, Kubernetes automatically assigns a memory request that matches the limit.

If you specify a limit but no request, then the limit IS the request.
If you specify a request but no limit, then there is no limit; the pod can use as much as it wants although it may be throttled or killed.
If you don't specify either, then there is nothing - there are no resources reserved for the pod, but it can use whatever it wants although it may be throttled or killed.

**Comment by zhoub** [comment]
Thanks for your reply, it's clear enough now.
